File: backend/app/schemas/conversation.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ConversationCreate(BaseModel):
    title: str | None = None


class ConversationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str | None
    summary: str | None
    created_at: datetime
    updated_at: datetime


class MessageResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    conversation_id: int
    role: str
    content: str
    metadata_: str | None
    created_at: datetime


class ChatRequest(BaseModel):
    conversation_id: int | None = None
    content: str

File: frontend/src/types/conversation.ts
export type MessageRole = 'user' | 'assistant' | 'system';

export interface Conversation {
  id: number;
  title: string;
  summary: string | null;
  created_at: string;
  updated_at: string;
}

export interface Message {
  id: number;
  conversation_id: number;
  role: MessageRole;
  content: string;
  metadata: Record<string, unknown> | null;
  created_at: string;
}

export interface ConversationCreate {
  title?: string;
}

export interface MessageCreate {
  content: string;
}
